### ISSUE TITLE
Song API discussion

### ISSUE THREAD

**Issue by alex-nexus** [jmansor/ipsy-music#3]
what would happen when attempting to updating songs to an album that doesn't belong to the artist?

**Comment by jmansor** [comment]
Good point, I completely missed this check. I think it can be redundant to have the artist associated to the song, as you can get it from the album, but in such case we should assume that every song will belong to an album, seems a too strong constraint, there could be songs from independent artists with no album. In any case, I don't figure how this could be checked from the database, so I would go with a validation by code, to make sure that the album belongs to the song's artist as well. A similar validations should be done when the artist is edited, although in this case, it seems right to just null the album.
